addons/libs/extdata.lua

```diff
@@ -1972,88 +1972,88 @@ end
-            [6] = "Apkallu Breeding",
-            [7] = "Wamoura Farm Raid",
-            [8] = "Egg Conservation",
-            [9] = "Operation: Black Pearl",
-            [10] = "Better than One",
+            [1] = {name="Excavation Duty"},
+            [2] = {name="Lebros Supplies"},
+            [3] = {name="Troll Fugitives"},
+            [4] = {name="Evade and Escape"},
+            [5] = {name="Siegemaster Assassination"},
+            [6] = {name="Apkallu Breeding"},
+            [7] = {name="Wamoura Farm Raid"},
+            [8] = {name="Egg Conservation"},
+            [9] = {name="Operation: Black Pearl"},
+            [10] = {name="Better than One"},
         },
-        [2494] = T{
+        [2494] = {
             -- "Periqia Diary"
-            [1] = "Seagull Grounded",
-            [2] = "Requiem",
-            [3] = "Saving Private Ryaaf",
-            [4] = "Shooting Down the Baron",
-            [5] = "Building Bridges",
-            [6] = "Stop the Bloodshed",
-            [7] = "Defuse the Threat",
-            [8] = "Operation: Snake Eyes",
-            [9] = "Wake the Puppet",
-            [10] = "The Price Is Right",
+            [1] = {name="Seagull Grounded"},
+            [2] = {name="Requiem"},
+            [3] = {name="Saving Private Ryaaf"},
+            [4] = {name="Shooting Down the Baron"},
+            [5] = {name="Building Bridges"},
+            [6] = {name="Stop the Bloodshed"},
+            [7] = {name="Defuse the Threat"},
+            [8] = {name="Operation: Snake Eyes"},
+            [9] = {name="Wake the Puppet"},
+            [10] = {name="The Price Is Right"},
         },
-        [2495] = T{
+        [2495] = {
             -- "Ilrusi Ledger"
-            [1] = "Golden Salvage",
-            [2] = "Lamia No.13",
-            [3] = "Extermination",
-            [4] = "Demolition Duty",
-            [5] = "Searat Salvation",
-            [6] = "Apkallu Seizure",
-            [7] = "Lost and Found",
-            [8] = "Deserter",
-            [9] = "Desperately Seeking Cephalopods",
-            [10] = "Bellerophon's Bliss",
+            [1] = {name="Golden Salvage"},
+            [2] = {name="Lamia No.13"},
+            [3] = {name="Extermination"},
+            [4] = {name="Demolition Duty"},
+            [5] = {name="Searat Salvation"},
+            [6] = {name="Apkallu Seizure"},
+            [7] = {name="Lost and Found"},
+            [8] = {name="Deserter"},
+            [9] = {name="Desperately Seeking Cephalopods"},
+            [10] = {name="Bellerophon's Bliss"},
         },
     }
 
     local names = missions[itemid]
-    local flags = {str:unpack("<q10")}
-    local data = T{flags}:key_map(function(id) return names[id] end)    
+    local flags = T{str:unpack("<q10")}
+    for key, val in ipairs(flags) do
+        names[key].completed = val
+    end
+
     local rettab = {
         type='Assault Log',
-        completed = S{data},
+        completed = names,
         flags = flags,
     }
```

Comment:
Not too happy with the structure of the result table. I would return something like this:
```lua
return {
    type = 'Assault Log',
    area = 'Ilrusi', -- will need to be looked up based on ID
    log = {
        { name = 'Golden Salvage', completed = false },
        ...
    },
}
```

With my above suggestion, this could be:
```lua
return {
    type = 'Assault Log',
    area = missions[item_id].name,
    log = table.range(10):map(function(index)
        return {
            name = missions[item_id][index],
            completed = flags[index],
        }
    end),
}
```